LabApiUtilities/src/test/com/microsoft/identity/labapi/utilities/authentication/TokenParametersTest.java

```diff
@@ -0,0 +1,126 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities.authentication;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A test to validate that we can create instances of {@link TokenParameters}.
+ */
+public class TokenParametersTest {
+
+    private final static String CLIENT_ID = "some_client_id";
+    private static final String AUTHORITY = "some_authority";
+    private static final String SCOPE_1 = "some_scope_1";
+    private static final String SCOPE_2 = "some_scope_2";
+
+    @Test(expected = NullPointerException.class)
+    public void testCannotCreateTokenParametersWithoutAuthority() {
+        TokenParameters.builder()
+                .clientId(CLIENT_ID)
+                .scope(SCOPE_1)
+                .build();
+
+        Assert.fail("We weren't expecting to hit this line...exception should've already occurred");
+    }
+
+    @Test
+    public void testTokenParametersScopesAreEmptyIfNotProvided() {
+        final TokenParameters tokenParameters = TokenParameters.builder()
+                .clientId(CLIENT_ID)
+                .authority(AUTHORITY)
+                .build();
+
+        Assert.assertNotNull(tokenParameters);
+        Assert.assertTrue(tokenParameters.getScopes().isEmpty());
+    }
+
+
+    @Test(expected = NullPointerException.class)
+    public void testCannotCreateTokenParametersWithoutClientId() {
+        TokenParameters.builder()
+                .authority(AUTHORITY)
+                .scope(SCOPE_1)
+                .build();
+
+        Assert.fail("We weren't expecting to hit this line...exception should've already occurred");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testCanCreateTokenParametersWhenRequiredDataProvided() {
+        final TokenParameters tokenParameters = TokenParameters.builder()
+                .authority(AUTHORITY)
+                .scope(SCOPE_1)
+                .build();
+
+        Assert.assertNotNull(tokenParameters);
+        Assert.assertEquals(AUTHORITY, tokenParameters.getAuthority());
+        Assert.assertEquals(CLIENT_ID, tokenParameters.getClientId());
+        Assert.assertNotNull(tokenParameters.getScopes());
+        Assert.assertEquals(1, tokenParameters.getScopes().size());
+        Assert.assertTrue(tokenParameters.getScopes().contains(SCOPE_1));
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
Where do we expect the NPE to come from?  There's a lot of extra lines in this test, likely.  Similar for others.

Reply:
yes, thanks for caching that. Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1389/commits/6ca827c98f3109498a735ee0a2f196f3423627e2